src/main/java/seedu/contax/ui/onboarding/OnboardingUtil.java

```diff
@@ -0,0 +1,141 @@
+            + "\n\nExample: add n/Johnny p/91234567 e/Johnny@j.com a/Johnny street";
+
+
+    /**
+     * Populates the given UniquePersonList with the sample data
+     * @param persons the UniquePersonList to be populated
+     */
+    public static void populateWithSample(UniquePersonList persons) {
+        Person[] samplePersons = SampleDataUtil.getSamplePersons();
+        for (Person p : samplePersons) {
+            persons.add(p);
+        }
+    }
+
+    /**
+     * Returns the name string of the last person in the given UniquepersonList
+     * @param persons the UniquePersonList to be searched
+     * @return name of the last person
+     */
+    public static String getLastestPersonName(UniquePersonList persons) {
+        ObservableList<Person> personObservableList = persons.asUnmodifiableObservableList();
+        String newPersonName = personObservableList.get(personObservableList.size() - 1).getName().toString();
+        return newPersonName;
+    }
+
+    /**
+     * Returns the last Person in the given UniquePersonList
+     * @param persons the UniquePersonList to be searched
+     * @return Person object of the last person
+     */
+    public static Person getLatestPerson(UniquePersonList persons) {
+        ObservableList<Person> personObservableList = persons.asUnmodifiableObservableList();
+        Person person = personObservableList.get(personObservableList.size() - 1);
+        return person;
+    }
+
+    /**
+     * Checks if the given command is valid. Updates the label with error message if it is not.
+     * @param command command string to be validated
+     * @param instructionLabel label to print error message on
+     */
+    public static boolean isCommandValid(String command, OnboardingInstruction instructionLabel) {
+        try {
+            parser.parseCommand(command);
+            return true;
+        } catch (ParseException e) {
+            instructionLabel.setText(INVALID_COMMAND);
+            return false;
+        }
+    }
+
+    /**
+     * Tokenizer to be used for onboaring guide. Tokenizes the given command based on the given Onboarding step.
+     * @param step OnboardingStep to tokenize command on
+     * @param command command string to be tokenized
+     * @return
+     */
+    public static ArgumentMultimap tokenize(OnboardingStep step, String command) {
+        if (step.getOperationId() == 6) {
+            return ArgumentTokenizer.tokenize(command, PREFIX_NAME, PREFIX_PHONE,
+                    PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+        }
+        return ArgumentTokenizer.tokenize(command);
+    }
+
+    /**
+     * Processes a given user command
+     * @param step
+     * @param command
+     * @param instructionLabel
+     * @param persons
+     * @return
+     */
+    public static int processCommand(OnboardingStep step, String command,
+                                     OnboardingInstruction instructionLabel, UniquePersonList persons) {
```

Comment:
You should not use UniquePersonList, its fairly low level in abstraction.
```suggestion
                                     OnboardingInstruction instructionLabel, Model model) {
```

Reply:
Updated accordingly